### DISCUSSION TITLE
Not able to commit my files

### DISCUSSION THREAD

**Discussion by akadcoder** [community/community#126051] [New to GitHub]
I am new to github and unable to push my files to my repo. Can anyone help me in this giving proper steps!

**Comment by akathedeveloper** [comment]
Bro first see the status of your files using:
    ```git status```
Then add the files in red using:
```git add filename```
Then commit the files with a message using: 
```git commit -m "your message```
Then try to push the files on your repo using:
```git push origin main```

Hope this will work for you!!

**Reply by jda8818** [reply]
Unable to push local git files to empty newly created github repo 
four committed files in local command-line git repo. [.gitignore, README.md, LICENSE, code.py]  github repo fresh - no files added - clean
github repo: verified email, git email changed to numerically coded no-reply email with git config and verified' -- push & fetch urls OK
PERSONAL ACCESS TOKEN OBTAINED
git push -u origin main requests username and password - Authentication Fails

**Comment by Abhishek22K** [comment]
Prerequisites
Git Installed: Make sure you have Git installed on your computer. You can download it from [here](https://git-scm.com/).
GitHub Account: Ensure you have a GitHub account. If not, you can create one [here](https://github.com/).
Steps to Push Files to GitHub
1. Create a Repository on GitHub
Go to [GitHub](https://github.com/) and log in.
Click on the + icon in the top right corner and select New repository.
Enter a repository name, description (optional), choose whether it’s public or private, and initialize it with a README file (optional).
Click Create repository.
2. Clone the Repository to Your Local Machine
Go to your new repository on GitHub.
Click on the Code button and copy the URL (either HTTPS or SSH).
Open your terminal or command prompt.
Navigate to the directory where you want to clone the repository:

cd path/to/your/directory
Clone the repository using the URL you copied:

git clone https://github.com/your-username/your-repo-name.git
Replace your-username and your-repo-name with your GitHub username and repository name.
3. Add Your Files to the Repository
Navigate to the cloned repository directory:

cd your-repo-name
Add or create the files you want to push to GitHub.
4. Add and Commit Your Changes
Check the status of your repository to see the files you've added or changed:

git status
Add the files to the staging area:

git add .
This command stages all the changes in the repository. You can also stage specific files by replacing . with the file names.
Commit the changes with a message describing what you’ve done:

git commit -m "Your commit message"
5. Push Your Changes to GitHub
Push your changes to the remote repository on GitHub:

git push origin main
Replace main with the name of the branch you want to push to if it’s different.
Example Workflow

Clone the repository:


git clone https://github.com/your-username/your-repo-name.git
cd your-repo-name

Add your files:

touch example-file.txt
echo "Hello, GitHub!" > example-file.txt

Stage and commit your changes:

git add example-file.txt
git commit -m "Add example file"

Push the changes:
git push origin main
Troubleshooting Tips
Authentication Issues: If you encounter authentication issues, ensure that you are using the correct credentials. You might need to generate a [personal access token](https://github.com/settings/tokens) for HTTPS authentication.
Remote Already Exists Error: If you get an error saying the remote already exists, it means you have already set the remote URL. You can verify this with git remote -v.
